test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,529 @@
+            self.zone.id,
+            hypervisor=hypervisor
+        )
+        cmd.zoneid = self.zone.id
+        cmd.templateid = template.id
+        cmd.serviceofferingid = self.service_offering.id
+        return self.apiclient.deployVirtualMachine(cmd)
+
+    def update_vm(self, id, extra_config):
+        cmd = updateVirtualMachine.updateVirtualMachineCmd()
+        cmd.id = id
+        cmd.extraconfig = extra_config
+        return self.apiclient.updateVirtualMachine(cmd)
+
+    def stop_vm(self, id):
+        cmd = stopVirtualMachine.stopVirtualMachineCmd()
+        cmd.id = id
+        return self.apiclient.stopVirtualMachine(cmd)
+
+    def start_vm(self, id):
+        cmd = startVirtualMachine.startVirtualMachineCmd()
+        cmd.id = id
+        return self.apiclient.startVirtualMachine(cmd)
+
+    # Parse extraconfig for config with that returned by xe vm-param-get ...
+    def get_xen_param_values(self, config):
+        equal_sign_index = config.index("=")
+        cmd_option = config[:equal_sign_index]
+        cmd_value = config[equal_sign_index + 1:]
+        return cmd_option, cmd_value
+
+    # Format vm config such that it equals the one from vmx file
+    def prepare_vmware_config(self, config):
+        equal_sign_index = config.index("=")
+        cmd_option = config[:equal_sign_index]
+        cmd_value = config[equal_sign_index + 1:]
+        return cmd_option + ' = '  '"{}"'.format(cmd_value)
+
+    # Get vm uuid from xenserver host
+    def get_vm_uuid(self, instance_name, ssh_client):
+        cmd = 'xe vm-list name-label={} params=uuid '.format(instance_name)
+        result = ssh_client.execute(cmd)
+        uuid_str = result[0]
+        i = uuid_str.index(":")
+        return uuid_str[i + 1:].strip()
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_01_deploy_vm_with_extraconfig_throws_exception_kvm(self):
+        '''
+        Test that extra config is not added when element tag is not added on the allowed list global config on KVM hosts
+        '''
+
+        hypervisor = self.hypervisor.lower()
+        if hypervisor != 'kvm':
+            raise self.skipTest("Skipping test case for non-kvm hypervisor")
+
+        '''
+        The following extraconfig is required for enabling hugepages on kvm
+        <memoryBacking>
+            <hugepages/>
+        </memoryBacking>
+        url encoded extra_config = '%3CmemoryBacking%3E%0D%0A++%3Chugepages%2F%3E%0D%0A%3C%2FmemoryBacking%3E'
+        '''
+        extraconfig = "%3CmemoryBacking%3E%0D%0A++%3Chugepages%2F%3E%0D%0A%3C%2FmemoryBacking%3E"
+
+        try:
+            # Clear KVM allow list to show that code throws exception when command is not included in the list
+            name = 'allow.additional.vm.configuration.list.kvm'
+
+            self.add_global_config(name, "")
+            self.assertRaises(Exception,
+                              self.deploy_vm(hypervisor, extraconfig),
```

Comment:
![Screenshot 2019-11-28 at 14 13 53](https://user-images.githubusercontent.com/13551960/69805459-59ced200-11e9-11ea-8cab-9aec5008a65c.png)
VMs per each hypervisor are being left behind, perhaps we could add a 'self.destroy_vm(response.id)' same as next tests